services/business/checkers/virtual_service_checker.go

```diff
@@ -0,0 +1,71 @@
+package checkers
+
+import (
+	"github.com/kiali/kiali/kubernetes"
+	"github.com/kiali/kiali/services/business/checkers/virtual_services"
+	"github.com/kiali/kiali/services/models"
+)
+
+const virtualCheckerType = "virtualservice"
+
+type VirtualServiceChecker struct {
+	Namespace        string
+	DestinationRules []kubernetes.IstioObject
+	VirtualService   []kubernetes.IstioObject
+}
+
+// An Object Checker runs all checkers for an specific object type (i.e.: pod, route rule,...)
+// It run two kinds of checkers:
+// 1. Individual checks: validating individual objects.
+// 2. Group checks: validating behaviour between configurations.
+func (in VirtualServiceChecker) Check() models.IstioValidations {
+	validations := models.IstioValidations{}
+
+	validations = validations.MergeValidations(in.runIndividualChecks())
+	validations = validations.MergeValidations(in.runGroupChecks())
+
+	return validations
+}
+
+// Runs individual checks for each virtual service
+func (in VirtualServiceChecker) runIndividualChecks() models.IstioValidations {
+	validations := models.IstioValidations{}
+
+	for _, virtualService := range in.VirtualService {
+		in.runChecks(virtualService, validations)
```

Comment:
In the no_service_checker we introduced to run the individual checkers concurrently.
I guess it could be a good opportunity to add it here, too.

Reply:
I get back to the PR with all data-races made by @simonpasquier and I do remember that he removed some concurrency in the exact same place but in route_rule_checker.go: https://github.com/kiali/kiali/pull/225/files#diff-14de00b9c8bae60d021a351f4d1070fbR33

I think this was because that code (and this one) is already executed within a go routine:
https://github.com/kiali/kiali/blob/master/services/business/istio_validations.go#L130

Does it make sense?

Reply:
istio_validations.go#runObjectCheckers() performs the validations concurrently.
And no_service_checker.go performs inside the Check() all the operations concurrently too.
I think it can follow same design in virtual_service_checker.go, too.

Reply:
I'll do the changes here: https://issues.jboss.org/browse/KIALI-1081

Reply:
IIRC I removed the goroutines because it seemed to me that the checks themselves didn't do any I/O (eg calls to the Istio/K8S APIs). In this case, spawning multiple goroutines wouldn't necessarily speed up things because the runtime has to switch contexts while no goroutine is actually idle.